### PR TITLE
feat: add LeetCode problem 434

### PR DESCRIPTION
**Pull request by PalAditya**
Adds solution of problem 434 for leetcode. Beats 100% Time, 97.18% space

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Comment by PalAditya** [comment]
TimeSpace

Proof of it being optimal 😄

**Review by Panquesito7 — DISMISSED** [review]
Thanks! 🚀

File: leetcode/src/434.c [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Given a string s, returns the number of segments in the string.
+int countSegments(char *s) {
+    int i, l = strlen(s), segments = 0;
+    if (l == 0) { // Return early for empty string
```
Comment (alexpantyukhin) [inline leetcode/src/434.c:4]: do we need this guard condition? I think if no, theresult will be still 0 in the end.

**Review by alexpantyukhin — COMMENTED** [review]
What if to simplify the solution with only one loop? Something like that:

```c
int countSegments(char * s){
    int sLen = strlen(s);
    int prevSpace = 1;
    int result = 0;
    char currChar;

    for (int i = 0; i < sLen; i++){
        currChar = s[i];
        if (s[i] != ' ' && prevSpace){
            result++;
        }

        prevSpace = (currChar == ' ');
    }

    return result;
}
```

It beats 100% :)

File: leetcode/src/434.c (hunk 1 above)
Comment (PalAditya) [reply leetcode/src/434.c]: Yep, you're right

**Comment by PalAditya** [comment]
@alexpantyukhin - Your version is simply superior, so I can't think of a good reason to not use it directly 😄 
I have added one explanatory comment which maybe unnecessary, but hopefully still explains the correctness of the logic as I see it.
P.S: Not sure if this PR is still valid, since I have pretty much copy-pasted your version apart from the comment. Please let me know

**Review by Panquesito7 — APPROVED** [review]
> Not sure if this PR is still valid, since I have pretty much copy-pasted your version apart from the comment. Please let me know

Both your contribution and @alexpantyukhin's contribution will be mentioned and kept, so I don't think there's much to worry about :) Thanks.

**Pull request merged**